### PR TITLE
add NumpyData.

### PR DESCRIPTION
**Pull request by AdamIzdebski**

### CODE REVIEW

**Comment by AdamIzdebski** [comment]
Tested and working

File: lso/data/numpy_data.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+
+@dataclass
+class NumpyData(lso_data.Data):
+    x: np.array
+    objective: Optional[np.array] = None
+    features: Optional[np.array] = None
+
+    def __add__(self, other: Any) -> "NumpyData":
+        if not isinstance(self.x, type(other.x)):
```
Comment (AdamIzdebski) [inline lso/data/numpy_data.py:16]: issubclass

File: lso/data/numpy_data.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            raise ValueError(f'cannot add {type(self.x)} to {type(other.x)}')
```
Comment (AdamIzdebski) [inline lso/data/numpy_data.py:17]: return NotImplemented

File: lso/data/numpy_data.py (hunk 1 above)
Comment (AdamIzdebski) [reply lso/data/numpy_data.py:16]: issubclass(other.x, type(self.x))

File: lso/data/numpy_data.py (hunk 2 above)
Comment (AdamIzdebski) [reply lso/data/numpy_data.py]: done

File: lso/data/numpy_data.py (hunk 1 above)
Comment (AdamIzdebski) [reply lso/data/numpy_data.py]: done

File: lso/data/numpy_data.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    x: np.array
+    objective: Optional[np.array] = None
+    features: Optional[np.array] = None
+
+    def __add__(self, other: Any) -> "NumpyData":
+
+        if not issubclass(type(other.x), type(self.x)):
+            return NotImplemented
+
+        if not issubclass(type(other.objective), type(self.objective)):
```
Comment (marmarmarmar) [inline lso/data/numpy_data.py:21]: You might have problems in case when one of these objects is `None`

**Review by AdamIzdebski — COMMENTED** [review]
Super!

**Pull request merged**